### PR TITLE
Fix issue with unknown calculation where value was not rounded

### LINKED ISSUE #1518: Calculation results: round results to 2 digits
OJ: Table 4a, mistake generated when the unknown ownership (4a) is using values from NDP in table 1a. Albania, Azerbaijan, Kazakhstan had a problem with the decimals. The NDP in table 1a generates a number of decimals which are not reflected in table 4a resulting in a very, very tiney negative difference. Had to add zeros manually in the sheet “Data”  (SEE ALBANIA: FIX: Round final calculation results)

### PR DESCRIPTION
**Pull request by sorja**
resolves #1518

before
-0.00 (value: 0.0025)

image

after
image

### CODE REVIEW

File: src/test/sectionSpec/fraSpecs.ts [hunk 1]
```diff
@@ -6025,7 +6025,7 @@ export const FraSpecs: Record<string, SectionSpec> = {
                 migration: {
                   colNames: ['1990', '2000', '2010', '2015'],
                   calcFormula: `forestOwnership.private_ownership || forestOwnership.public_ownership 
-                    ? extentOfForest.forestArea - (forestOwnership.private_ownership || 0) - (forestOwnership.public_ownership || 0)
+                    ? Math.round((extentOfForest.forestArea - (forestOwnership.private_ownership || 0) - (forestOwnership.public_ownership || 0) + Number.EPSILON) * 100) / 100
```
Comment (minotogna) [inline src/test/sectionSpec/fraSpecs.ts]: all calculation results must be rounded to 2. not each single formula.
check src/server/controller/cycleData/persistNodeValue/calculateDependantNodes/calculateNode.ts line 37 where results is parsed from ExpressionEvaluator

**Review by minotogna — CHANGES_REQUESTED** [review]

File: src/test/sectionSpec/fraSpecs.ts (hunk 1 above)
Comment (sorja) [reply src/test/sectionSpec/fraSpecs.ts]: great, wasn't 100% sure if this was what we wanted, ill fix this so

File: src/server/controller/cycleData/persistNodeValue/calculateDependantNodes/calculateNode.ts [hunk 2]
```diff
@@ -34,7 +36,10 @@ export const calculateNode = async (
     row,
     formula: expression,
   })
-  const value: NodeValue = { raw: rawResult ? String(rawResult) : null, calculated: true }
+
+  const roundedResult = Numbers.format(Numbers.toBigNumber(rawResult))
+
+  const value: NodeValue = { raw: rawResult ? roundedResult : null, calculated: true }
```
Comment (minotogna) [inline src/server/controller/cycleData/persistNodeValue/calculateDependantNodes/calculateNode.ts]: I would round if there's the rawResult , not necessarily always

Comment (sorja) [reply src/server/controller/cycleData/persistNodeValue/calculateDependantNodes/calculateNode.ts]: Makes sense!

**Review by rfontanarosa — APPROVED** [review]

**Review by minotogna — APPROVED** [review]

**Pull request merged**